misc-scripts/xref_mapping/XrefParser/DBASSParser.pm

```diff
@@ -20,124 +20,185 @@ limitations under the License.
+      my ( $first_gene_name, $second_gene_name );
+      if ( ( $dbass_gene_name =~ m{
+                                    (.*)
+                                    \s?\/\s?  # typically no ws here but just in case
+                                    (.*)
+                                  }msx ) ||
+           ( $dbass_gene_name =~ m{
+                                    (.*)
+                                    \s?  # typically there IS a space before the ( here
+                                    [(] (.*) [)]
+                                  }msx ) ) {
+        $first_gene_name  = $1;
+        $second_gene_name = $2;
+      }
+      else {
+        $first_gene_name = $dbass_gene_name;
+        $second_gene_name = undef;
+      }
+
+      my $label       = $first_gene_name;
+      my $synonym     = $second_gene_name;
+      my $type        = 'gene';
+      my $version     = '1';
+
+      my $xref_id =
+        $self->get_xref( $dbass_gene_id, $source_id, $species_id, $dbi );
+
+      if ( !defined($xref_id) || $xref_id eq q{} ) {
+        $xref_id = $self->add_xref(
+                                   { acc        => $dbass_gene_id,
+                                     version    => $version,
+                                     label      => $label,
+                                     desc       => undef,
+                                     source_id  => $source_id,
+                                     dbi        => $dbi,
+                                     species_id => $species_id,
+                                     info_type  => "DIRECT" } );
+      }
+
+      if ( defined($synonym) ) {
+        $self->add_synonym( $xref_id, $synonym, $dbi );
+      }
+
+      $self->add_direct_xref( $xref_id, $ensembl_id, $type, undef, $dbi );
+
+      ++$processed_count;
+    }
+    else {
+      ++$unmapped_count;
+    }
 
+  } ## end while ( defined( my $line...))
 
-    return 0;
+  $csv->eof || croak 'Error parsing CSV: ' . $csv->error_diag();
+  $file_io->close();
+
+  if ($verbose) {
+    printf( "%d direct xrefs succesfully processed\n", $processed_count );
+    printf( "Skipped %d unmapped xrefs\n", $unmapped_count );
+  }
+
+  return 0;
 } ## end sub run
 
 
+=head2 is_file_header_valid
+
+  Arg [1]    : String file header line
+  Example    : if (!is_file_header_valid($header_line)) {
+                 croak 'Bad header';
+               }
+  Description: Verifies if the header of a DBASS file follows expected
+               syntax and contains expected column names.
+  Return type: boolean
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub is_file_header_valid {
+  my ( $header ) = @_;
+
+  # Don't bother with parsing column names if their number does not
+  # match to begin with
+  if ( scalar @{ $header } != $EXPECTED_NUMBER_OF_COLUMNS ) {
+    return 0;
+  }
+
+  my @fields_ok;
+
+  my ( $dbass_end ) = ( $header->[0] =~ m{ DBASS (3|5) GeneID }msx );
+  push @fields_ok, defined $dbass_end;
+
```

Comment:
Why the variable is named as $dbass_end?  As it is capturing the header (DBASS3GeneID/DBASS5GeneID), could be named as $dbass_geneid or $dbass_geneid_ok for consistency with other variable names below.

Also 'defined' check is done only for the first field and not for other fields?!

Reply:
Good. Well explained.